### ISSUE TITLE
Istio 1.0.4 Helm chart Grafana persistence errors

### ISSUE THREAD

**Issue by ewbankkit** [istio/istio#10179]
Deploying Istio 1.0.4 via Helm chart with
```yaml
grafana:
  persist: true
```
I get
```
Release "istio" does not exist. Installing it now.
Error: release istio failed: Deployment.apps "grafana" is invalid: spec.template.spec.containers[0].volumeMounts[1].name: Not found: "dashboards-istio"
```

If I massage the chart's `deployment.yaml` so that the Grafana pod starts then the Grafana pod crashes:
```
GF_PATHS_DATA='/data/grafana' is not writable.
You may have issues with file permissions, more information here: http://docs.grafana.org/installation/docker/#migration-from-a-previous-version-of-the-docker-container-to-5-1-or-later
t=2018-11-28T20:46:36+0000 lvl=info msg="Starting Grafana" logger=server version=5.2.3 commit=cf4c090 compiled=2018-08-22T08:20:39+0000
t=2018-11-28T20:46:36+0000 lvl=info msg="Config loaded from" logger=settings file=/usr/share/grafana/conf/defaults.ini
t=2018-11-28T20:46:36+0000 lvl=info msg="Config loaded from" logger=settings file=/etc/grafana/grafana.ini
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.data=/data/grafana"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.logs=/var/log/grafana"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.plugins=/var/lib/grafana/plugins"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.provisioning=/etc/grafana/provisioning"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.log.mode=console"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_DATA=/data/grafana"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_LOGS=/var/log/grafana"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_PLUGINS=/var/lib/grafana/plugins"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_PROVISIONING=/etc/grafana/provisioning"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_ANONYMOUS_ENABLED=true"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_ANONYMOUS_ORG_ROLE=Admin"
t=2018-11-28T20:46:36+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_AUTH_BASIC_ENABLED=false"
t=2018-11-28T20:46:36+0000 lvl=info msg="Path Home" logger=settings path=/usr/share/grafana
t=2018-11-28T20:46:36+0000 lvl=info msg="Path Data" logger=settings path=/data/grafana
t=2018-11-28T20:46:36+0000 lvl=info msg="Path Logs" logger=settings path=/var/log/grafana
t=2018-11-28T20:46:36+0000 lvl=info msg="Path Plugins" logger=settings path=/var/lib/grafana/plugins
t=2018-11-28T20:46:36+0000 lvl=info msg="Path Provisioning" logger=settings path=/etc/grafana/provisioning
t=2018-11-28T20:46:36+0000 lvl=info msg="App mode production" logger=settings
t=2018-11-28T20:46:36+0000 lvl=info msg="Initializing SqlStore" logger=server
t=2018-11-28T20:46:36+0000 lvl=info msg="Connecting to DB" logger=sqlstore dbtype=sqlite3
t=2018-11-28T20:46:36+0000 lvl=info msg="Starting DB migration" logger=migrator
t=2018-11-28T20:46:36+0000 lvl=eror msg="Server shutdown" logger=server reason="Service init failed: Migration failed err: unable to open database file"
```

**Comment by Stono** [comment]
I'm not sure about the fix that was merged here.
Making it configurable is fine, but changing from ReadWriteOnce to ReadWriteMany is not a storage mode supported by any of the cloud provider default storageClasses (eg, gcp persistent disks etc) and is only supported if you're running a specific distributed shared storage.

**Comment by ewbankkit** [comment]
@Stono I copied from https://github.com/istio/istio/pull/8416 which is in Istio 1.1; I found that I had to override with `ReadWriteOnce` for Amazon EBS FWIW. I guess the thing to do would be to open an issue against the 1.1 code.

**Comment by ewbankkit** [comment]
Fix merged to `release-1.0` with https://github.com/istio/istio/pull/10180.
